packages/expect/src/toThrowMatchers.js

```diff
@@ -7,203 +7,303 @@
  * @flow
  */
 
-import type {MatchersObject} from 'types/Matchers';
+import type {MatcherHintOptions, MatchersObject} from 'types/Matchers';
 
-import getType from 'jest-get-type';
-import {escapeStrForRegex} from 'jest-regex-util';
 import {formatStackTrace, separateMessageFromStack} from 'jest-message-util';
 import {
   EXPECTED_COLOR,
   RECEIVED_COLOR,
-  highlightTrailingWhitespace,
   matcherErrorMessage,
   matcherHint,
   printExpected,
   printReceived,
   printWithType,
 } from 'jest-matcher-utils';
-import {equals} from './jasmineUtils';
 import {isError} from './utils';
 
-export const createMatcher = (matcherName: string, fromPromise?: boolean) => (
-  actual: Function,
-  expected: string | Error | RegExp,
-) => {
-  const value = expected;
-  let error;
-
-  if (fromPromise && isError(actual)) {
-    error = actual;
-  } else {
-    if (typeof actual !== 'function') {
-      if (!fromPromise) {
-        throw new Error(
-          matcherErrorMessage(
-            matcherHint('[.not]' + matcherName, undefined, undefined),
-            `${RECEIVED_COLOR('received')} value must be a function`,
-            printWithType('Received', actual, printReceived),
-          ),
-        );
+const DID_NOT_THROW = 'Received function did not throw an exception';
```

Comment:
Similarly to @SimenB's

> "thrown value" over "exception value",

I think this should rather be
```suggestion
const DID_NOT_THROW = 'Received function did not throw';
```
because it did not throw `42`, `undefined`, `window` or something else that's not an error either 😄

Reply:
Yes, less is more :)